src/cartesian/Area.js

```diff
@@ -199,10 +199,10 @@ class Area extends Component {
 
   state = { isAnimationFinished: true };
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
```

Comment:
In most of your `componentDidUpdate` changes you're using the wrong props. I haven't checked them all, but I'll use the first one as an example.

This method wants to cache the old props, but since you've blindly translated `nextProps` to `prevProps`, this method now caches thew *new* props.

What you should do is destruct `prevProps` instead of `this.props`, then use `this.props` instead of `prevProps` in the conditional.

Reply:
Have so much sense mate, so tired yesterday.
I'll fix the pull request asap